shuffle/cli/src/account.rs

```diff
@@ -26,20 +28,48 @@ use std::{
     io,
     path::{Path, PathBuf},
 };
+use url::Url;
 
 // Creates new account from randomly generated private/public key pair.
-pub fn handle(home: &Home, root: Option<PathBuf>, network: Network) -> Result<()> {
+pub async fn handle(home: &Home, root: Option<PathBuf>, network: Network) -> Result<()> {
     let network_home = home.new_network_home(&network.get_name());
     network_home.generate_paths_if_nonexistent()?;
     check_nodeconfig_exists_if_localhost_used(home, &network)?;
 
-    if network_home.get_latest_account_path().exists() {
+    if network_home.get_latest_account_key_path().exists() {
         match user_wants_another_key(&network_home) {
             true => archive_current_files_in_latest(&network_home)?,
             false => return Ok(()),
         }
     }
+    let new_account = generate_new_account(&network_home)?;
+    let test_account = generate_test_account(&network_home)?;
+
+    //creating account onchain based on network provided by user
+    match network.get_name().as_str() {
+        LOCALHOST_NAME => handle_account_creation_on_localhost(
+            home,
+            &network_home,
+            &network,
+            root,
+            &new_account,
+            &test_account,
+        ),
+        TROVE_TESTNET_NETWORK_NAME => {
+            handle_account_creation_on_network(&network, &new_account, &test_account).await
```

Comment:
Both `handle_account_creation_on_localhost ` and `handle_account_creation_on_network` could use better names imo.

1. Let's change this method name to accurately reflect that it's using a faucet, rather than merely on a network.
2. We could still run `handle_account_creation_on_localhost` on a network as long as we had the right mint.key, so perhaps `handle_account_creation_on_devapi`, or something similar is better.

Reply:
changed to `create_account_via_dev_api` and `create_account_via_faucet`